### PR TITLE
fix(Asana Node): Fix issue when connecting to the new Asana environment

### PR DESCRIPTION
**Pull request by Joffcom**
## Summary
This fixes an issue that was preventing the Asana and Asana Trigger node from connecting to the updated Asana environment.

## Related tickets and issues
https://github.com/n8n-io/n8n/issues/8403
https://community.n8n.io/t/asana-returns-error-403-for-get-requests-after-implementing-cdn/35605

### CODE REVIEW

File: packages/nodes-base/nodes/Asana/GenericFunctions.ts [hunk 1]
```diff
@@ -29,10 +29,16 @@ export async function asanaApiRequest(
 		method,
 		body: method === 'GET' || method === 'HEAD' || method === 'DELETE' ? null : { data: body },
 		qs: query,
-		url: uri || `https://app.asana.com/api/1.0${endpoint}`,
+		url:
+			uri ||
+			`https://app.asana.com/api/1.0/${endpoint.startsWith('/') ? endpoint.slice(1) : endpoint}`,
```
Comment (netroy) [inline packages/nodes-base/nodes/Asana/GenericFunctions.ts:34]: we should instead change the `endpoint` type from `string` to `/${string}`, and that way this is caught at compile-time, and we don't need to maintain yet another line that conditionally removes `/`.

**Review by netroy — CHANGES_REQUESTED** [review]

File: packages/nodes-base/nodes/Asana/GenericFunctions.ts (hunk 1 above)
Comment (netroy) [reply packages/nodes-base/nodes/Asana/GenericFunctions.ts:34]: I've pushed my proposed changes [here](https://github.com/n8n-io/n8n/compare/gh/8403...8404-proposal?expand=1). If they look acceptable, please feel free to cherry-pick.

Comment (Joffcom) [reply packages/nodes-base/nodes/Asana/GenericFunctions.ts]: Checked the URIs and it looks like they are all ok anyway, I was just being cautious based on previous experience. Also spotted a typo and fixed that.

Comment (Joffcom) [reply packages/nodes-base/nodes/Asana/GenericFunctions.ts]: Ah it didn't update quick enough to show your changes, Will grab your changes now as they make more sense.

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.25.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.25.1)

**Pull request merged**